doc/modules/model_evaluation.rst

```diff
@@ -2403,34 +2407,92 @@ the difference in errors decreases. Finally, by setting, ``power=2``::
 we would get identical errors. The deviance when ``power=2`` is thus only
 sensitive to relative errors.
 
-.. _d2_tweedie_score:
+.. _d2_score:
 
-D² score, the coefficient of determination
--------------------------------------------
+D² score
+--------
 
-The :func:`d2_tweedie_score` function computes the percentage of deviance
-explained. It is a generalization of R², where the squared error is replaced by
-the Tweedie deviance. D², also known as McFadden's likelihood ratio index, is
-calculated as
+The D² score computes the fraction of deviance explained. 
+It is a generalization of R², where the squared error is generalized and replaced 
+by a deviance of choice :math:`\text{D}(y, \hat{y})`
+(e.g., Tweedie, pinball or mean absolute error).
+It is calculated as
 
 .. math::
 
   D^2(y, \hat{y}) = 1 - \frac{\text{D}(y, \hat{y})}{\text{D}(y, \bar{y})} \,.
 
-The argument ``power`` defines the Tweedie power as for
-:func:`mean_tweedie_deviance`. Note that for `power=0`,
-:func:`d2_tweedie_score` equals :func:`r2_score` (for single targets).
+Where :math:`\bar{y}` is the optimal intercept-only model's prediction 
+(e.g., the mean of `y_true` for the Tweedie case, the median for absolute 
+error and the alpha-quantile for pinball loss).
 
 Like R², the best possible score is 1.0 and it can be negative (because the
 model can be arbitrarily worse). A constant model that always predicts the
 expected value of y, disregarding the input features, would get a D² score
 of 0.0.
 
+D² Tweedie score
+^^^^^^^^^^^^^^^^
+
+The :func:`d2_tweedie_score` function implements the special case of D² 
+where :math:`\text{D}(y, \hat{y})` is the Tweedie deviance. 
+It is also known as D² Tweedie or McFadden's likelihood ratio index.
+
+The argument ``power`` defines the Tweedie power as for
+:func:`mean_tweedie_deviance`. Note that for `power=0`,
+:func:`d2_tweedie_score` equals :func:`r2_score` (for single targets).
+
 A scorer object with a specific choice of ``power`` can be built by::
 
   >>> from sklearn.metrics import d2_tweedie_score, make_scorer
   >>> d2_tweedie_score_15 = make_scorer(d2_tweedie_score, power=1.5)
 
+D² pinball score
+^^^^^^^^^^^^^^^^^^^^^
+
+The :func:`d2_pinball_score` function implements the special case
+of D² with a pinball deviance, i.e.:
+
+.. math::
+
+  D(y, \hat{y}) = \sum_{i=0}^{n_{\text{samples}}-1}  \alpha \max(y_i - \hat{y}_i, 0) + (1 - \alpha) \max(\hat{y}_i - y_i, 0).
+
+The argument ``alpha`` defines the slope of the pinball deviance as for
```

Comment:
Please use "pinball loss" instead of "pinball deviance".

Reply:
Thanks, fixed!